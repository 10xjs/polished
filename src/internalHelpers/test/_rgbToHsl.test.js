// @flow

import rgbToHsl from '../_rgbToHsl'

describe('hslToHex', () => {
  it('should convert the color red', () => {
    expect({ background: rgbToHsl(255, 0, 0) }).toMatchSnapshot()
  })

  it('should convert the color blue', () => {
    expect({ background: rgbToHsl(0, 0, 255) }).toMatchSnapshot()
  })

  it('should convert the color green', () => {
    expect({ background: rgbToHsl(0, 255, 0) }).toMatchSnapshot()
  })

  it('should convert black', () => {
    expect({ background: rgbToHsl(0, 0, 0) }).toMatchSnapshot()
  })
})
