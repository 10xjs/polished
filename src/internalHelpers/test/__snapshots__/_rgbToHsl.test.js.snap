exports[`hslToHex should convert black 1`] = `
Object {
  "background": Object {
    "hue": 0,
    "lightness": 0,
    "saturation": 0,
  },
}
`;

exports[`hslToHex should convert the color blue 1`] = `
Object {
  "background": Object {
    "hue": 240,
    "lightness": 0.5,
    "saturation": 1,
  },
}
`;

exports[`hslToHex should convert the color green 1`] = `
Object {
  "background": Object {
    "hue": 120,
    "lightness": 0.5,
    "saturation": 1,
  },
}
`;

exports[`hslToHex should convert the color red 1`] = `
Object {
  "background": Object {
    "hue": 0,
    "lightness": 0.5,
    "saturation": 1,
  },
}
`;
